package/MDAnalysis/core/selection.py

```diff
@@ -979,6 +1012,7 @@ def __init__(self, parser, tokens):
             raise ValueError(
                 "Invalid operator : '{0}' Use one of : '{1}'"
                 "".format(oper, self.ops.keys()))
+        self.value = float(value)
```

Comment:
Yep I did think along those lines.  I was also thinking if there'd be a neat way of implementing chained comparisons (`10 < x < 15`) by peeking at the next token and seeing if it's a operator.

So

```python
if tokens[0] in self.ops:
    # split this into two selections, checking if it logically makes sense...
    return AndOperation(firstsel, secondsel)
```